### PR TITLE
[#152932975] Refactor Trading Ticket account checks to use order capabilities

### PR DESCRIPTION
**Pull request by guillaumedebavelaere**

### CODE REVIEW

**Comment by jsom** [comment]
Now that we call the order capabilities every time the screen appears won't that slow the user down unnecessarily? If they're switching the order type, etc we don't need to call update order caps again. cc @mitochondrion as I believe you made this change on FX?

**Comment by guillaumedebavelaere** [comment]
Is there another way than passing a flag we could call orderCapabilities only if the account has changed?

File: TradeItIosTicketSDK2/TradeItTradingTicketViewController.swift [hunk 1]
```diff
@@ -81,46 +82,21 @@ class TradeItTradingTicketViewController: TradeItViewController, UITableViewData
...
-                        withTitle: "Unsupported Account",
-                        withMessage: "The selected account does not support trading stocks. Please choose another account.",
-                        withActionTitle: "OK",
-                        onAlertActionTapped: {
-                            self.delegate?.invalidAccountSelected(
-                                onTradingTicketViewController: self,
-                                withOrder: self.order
-                            )
-                        }
+        guard let _ = self.orderCapabilities else {
```
Comment (mitochondrion) [inline TradeItIosTicketSDK2/TradeItTradingTicketViewController.swift]: This doesn't seem to be checking if equity trading is supported, just if orderCapabilities exists, which won't exist if we didn't call authenticate yet.

File: TradeItIosTicketSDK2/TradeItYahooTradingTicketViewController.swift [hunk 2]
```diff
@@ -66,51 +65,7 @@ class TradeItYahooTradingTicketViewController: TradeItYahooViewController, UITab
...
-            }
-        )
-
-        self.reloadTicket()
-    }
-
-    override func viewDidAppear(_ animated: Bool) {
-        super.viewDidAppear(animated)
-        self.fireViewEventNotification(view: .trading, title: self.title)
+        self.updateOrderCapabilities()
```
Comment (mitochondrion) [inline TradeItIosTicketSDK2/TradeItYahooTradingTicketViewController.swift]: I thought we were going to separate this stuff so we don't fetch balances every time we come back to the screen...

Comment (guillaumedebavelaere) [reply TradeItIosTicketSDK2/TradeItYahooTradingTicketViewController.swift]: I haven't updated yet the Yahoo ticket. I wanted to confirm that was correct for the Trade it Ticket to update it.

File: TradeItIosTicketSDK2/TradeItTradingTicketViewController.swift (hunk 1 above)
Comment (guillaumedebavelaere) [reply TradeItIosTicketSDK2/TradeItTradingTicketViewController.swift]: This is set before. On `viewDidLoad` we call `updateOrderCapabilities` and in it: `self.orderCapabilities = (self.order.linkedBrokerAccount?.orderCapabilities.filter { $0.instrument == "equities" })?.first`
We set it also when we switch account.

**Comment by guillaumedebavelaere** [comment]
@jsom Yahoo Trading Ticket updated

**Comment by jsom** [comment]
Needs a rebase

**Comment by guillaumedebavelaere** [comment]
@jsom rebased

**Pull request merged**